addons/l10n_ar_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,149 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields, api, Command, _
+from odoo.exceptions import UserError
+
+
+class AccountPaymentRegisterWithholding(models.TransientModel):
+    _name = 'account.payment.register.withholding'
+    _description = 'account.payment.register.withholding'
+
+    payment_register_id = fields.Many2one('account.payment.register', required=True, ondelete='cascade',)
+    currency_id = fields.Many2one(related='payment_register_id.currency_id')
+    name = fields.Char(string='Number', required=False)
+    tax_id = fields.Many2one('account.tax', required=True,)
+    base_amount = fields.Monetary(required=True, compute='_compute_base_amount', store=True, readonly=False)
+    amount = fields.Monetary(required=True, compute='_compute_amount', store=True, readonly=False)
+
+    @api.depends('tax_id', 'payment_register_id.line_ids', 'payment_register_id.amount')
+    def _compute_base_amount(self):
+        # TODO improve this method and send some funcionality to account.tax, this approach is for demo purpose
```

Comment:
we cuuld remote thiscomment, remove the tax_amount use case and check multcurrency use case (I believe it's working properly)